pallets/roles/src/impls.rs

```diff
@@ -85,9 +84,9 @@ impl<T: Config> Pallet<T> {
 		address: T::AccountId,
 		slash_amount: T::CurrencyBalance,
 	) -> sp_runtime::DispatchResult {
-		let mut ledger = Self::ledger(&address).ok_or(Error::<T>::InvalidStashController)?;
+		let mut ledger = Self::ledger(&address).ok_or(Error::<T>::AccountNotPaired)?;
 		let (_imbalance, _missing) = T::Currency::slash(&address, slash_amount.into());
```

Comment:
In this case, how does the `staking` pallet know that the validator has been slashed?

Reply:
The slashing part is not covered.. will open separate PR for it.